src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -173,6 +177,16 @@ IObservableVector<Model::Profile> CascadiaSettings::ActiveProfiles() const noexc
     return _activeProfiles;
 }
 
+IVectorView<Model::ExtensionPackage> CascadiaSettings::Extensions()
+{
+    if (!_extensionPackages)
+    {
+        // Lazy load the ExtensionPackage objects

```

Comment:
FUNNY. I bet we only call this on the CLONED one, so the cache is always empty (clone the base settings object without the extensions, get the extensions, fill the cache)

I think this is fine, fwiw, because it results in users seeing up-to-date fragment information on every settings UI load ;p

Reply:
Sorta. When we `Copy()`, we completely copy over `_extensionPackages`. So we should lazy load them on the clone when we open the SUI the first time. Then "discard changes" should regenerate it (as you said) because `_settingsSource` hasn't loaded it. But --

Yeah nevermind, you're right. We could make it static, but meh, I agree with you, it's nice to see up-to-date fragment info.